src/index.html

```diff
@@ -10,8 +10,195 @@
+          Name:
+        </label>
+        <input
+          class="form-controls"
+          type="text"
+          autocomplete="off"
+          name="name"
+          minlength="2"
+          maxlength="30"
+          id="name"
+          required>
+        <br>
+        <label for="age">
+          How old are You?
+        </label>
+        <input
+          class="form-controls"
+          type="number"
+          value="12"
+          min="1"
+          max="100"
+          name="age"
+          id="age"
+          required>
+        <br>
+        <label for="dob">
+          Full date of birth:
+        </label>
+        <input
+          class="form-controls"
+          type="date"
+          name="dob"
+          id="dob"
+          required>
+        <br>
+        <label for="terms">
+          I accept the term of the agreement
+        </label>
+        <input
+          type="checkbox"
+          name="terms"
+          id="terms"
+          required>
+      </fieldset>
+
+      <fieldset class="field-group">
+        <legend>Registration</legend>
+        <label for="email">
+          E-mail:
+        </label>
+        <input
+          class="form-controls"
+          type="email"
+          placeholder="email@example.com"
+          name="email"
+          id="email"
+          required>
+        <br>
+        <label for="password">
+          Password:
+        </label>
+        <input
+          type="password"
+          name="password"
+          minlength="8"
+          id="password"
+          required
+        >
+      </fieldset>
+
+      <fieldset class="field-group">
+        <legend>An interesting fact about you!</legend>
+        <div class="form-controls">
+          <label for="fact">
+            Do you love cats?
+          </label>
+          <input
+            type="radio"
+            name="yes"
+            id="yes"
+            value="yes">
+          <label for="yes">
+            Yes
+          </label>
+          <input
+            type="radio"
+            name="no"
+            id="no"
+            value="no"
+          >
+          <label for="no">
+            No
+          </label>
+        </div>
+
+        <div class="form-controls">
+          <label for="color">
+            What is your favorite color?
+          </label>
+          <input
+            type="color"
+            id="color">
+        </div>
+
+        <div class="form-controls">
+          <label for="time">
+            What time do you go to bed?
+          </label>
+          <input
+            type="time"
+            id="time">
+        </div>
+
+        <div class="form-controls">
+          <label for="cars">
+            What are your favorite brand of cars?
+          </label>
+          <select
+            name="cars"
+            id="cars"
+            multiple
+          >
+            <option value="BMW">BMW</option>
+            <option value="Audi">Audi</option>
+            <option value="Lada">Lada</option>
```

Comment:
minor comment - when passing data to a server it's common to lower-case values to avoid letter-case-driven issues:
```suggestion
            <option value="bmw">BMW</option>
            <option value="audi">Audi</option>
            <option value="lada">Lada</option>
```